src/imu_full_measured_data_provider.cpp

```diff
@@ -0,0 +1,62 @@
+/***************************************************************************//**
+*   @file   imu_full_measured_data_provider.cpp
+*   @brief  Implementation for imu full measured data
+*   @author Vasile Holonec (Vasile.Holonec@analog.com)
+********************************************************************************
+* Copyright 2023(c) Analog Devices, Inc.
+
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "imu_ros2/imu_full_measured_data_provider.h"
+
+ImuFullMeasuredDataProvider::ImuFullMeasuredDataProvider()
+{
+    init();
+}
+
+ImuFullMeasuredDataProvider::~ImuFullMeasuredDataProvider()
+{
+
+}
+
+void ImuFullMeasuredDataProvider::init()
+{
+    // initialize a library
+}
+
+imu_ros2::msg::ImuFullMeasuredData ImuFullMeasuredDataProvider::getData(bool& success)
+{
+    success = true;
```

Comment:
why do you have a success variable if you do not verify if the readings are successful? the get APIs below should return an error status and success should be updated according to their status